classes/services/PKPSubmissionService.inc.php

```diff
@@ -296,6 +296,50 @@ public function getReviewAssignments($submission) {
 		return $reviewAssignmentDao->getBySubmissionId($submission->getId());
 	}
 
+	/**
+	 * Is this user allowed to view the author details?
+	 *
+	 * - Anyone can view published submission authors
+	 * - Reviewers can only view authors in open reviews
+	 * - Managers and admins can view authors of any submission
+	 * - Subeditors, authors and assistants can only view authors in assigned subs
+	 *
+	 * @param $user User
+	 * @param $submission Submission
+	 * @return boolean
+	 */
+	public function canUserViewAuthor($user, $submission) {
+
+		if ($submission->getDatePublished()) {
+			return true;
+		}
+
+		$reviewAssignments = $this->getReviewAssignments($submission);
+		foreach ($reviewAssignments as $reviewAssignment) {
+			if ($user->getId() == $reviewAssignment->getReviewerId()) {
+				return $reviewAssignment->getReviewMethod() == SUBMISSION_REVIEW_METHOD_OPEN ? true : false;
```

Comment:
Single blind should permit the reviewer to see the author's name too, no?

Reply:
Yeah? Ok, I didn't actually know what "single blind" means but I've looked it up now. :laughing:

Reply:
https://github.com/NateWr/pkp-lib/commit/e2362c1183d6b075d0ab9eea6e2e0762a57f4ce0